src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -355,13 +355,29 @@ public string DownloadFile(string storageKeyOfFile)
 		/// <param name="storageKeyOfBookFolder"></param>
 		public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinationParentDirectory, ProgressDialog downloadProgress = null)
 		{
-			//TODO tell it not to download pdfs. Those are just in there for previewing purposes, we don't need to get them now that we're getting the real thing
-
 			//review: should we instead save to a newly created folder so that we don't have to worry about the
 			//other folder existing already? Todo: add a test for that first.
 
-			if (!GetBookExists(storageKeyOfBookFolder))
+			// We need to download individual files to avoid downloading unwanted files (PDFs and thumbs.db to
```

Comment:
I've refactored the code with a couple of methods whose names should help somewhat.

Note that storageKeyOfBookFolder can be interpreted as a directory string with directories separated by  "/" characters, but is actually a monolithic key into a database table.  Likewise, the individual object keys behave the same way.  The individual object keys should all begin with the string from storageKeyOfBookFolder.  So the relative local filepaths can be obtained by removing the storageKeyOfBookFolder string from the beginning of the individual object key.

The basic algorithm is as follows:
1) Get the list of objects contained in the database for storageKeyOfBookFolder.
2) Check whether the list contains any desired files after filtering.  If not, quit.
3) Get the book's local folder name from the key of the first object. (any object's key will do)
4) Create a temporary directory and initialize the progress reporter.
5) For each desired object/file, download it to a location in the temporary directory based on its key.  (This handles subdirectory structure, if any.)
6) Copy/move the downloaded directory content to the desired final location in the filesystem.